python/orca/src/bigdl/orca/learn/pytorch/pytorch_pyspark_worker.py

```diff
@@ -126,32 +126,38 @@ def setup_distributed(self, mode, cluster_info, driver_ip, driver_tcp_store_port
     def train_epochs(self, data_creator, epochs=1, batch_size=32, profile=False,
                      wrap_dataloader=None, callbacks=None,
                      validation_data_creator=None):
-        self.load_state_dict(self.state_dict.value)
+        if self.state_dict:
+            self.load_state_dict(self.state_dict.value)
         stats_list = super().train_epochs(data_creator=data_creator,
                                           epochs=epochs,
                                           batch_size=batch_size,
                                           profile=profile,
                                           wrap_dataloader=wrap_dataloader,
                                           callbacks=callbacks,
                                           validation_data_creator=validation_data_creator)
-        state_dict = self.get_state_dict()
 
         if self.log_to_driver:
             LogMonitor.stop_log_monitor(self.log_path, self.logger_thread, self.thread_stop)
 
         if self.rank == 0:
             if self.model_dir is not None:
+                state_dict = self.get_state_dict()
                 save_pkl(state_dict, os.path.join(self.model_dir, "state.pkl"))
 
         if self.model_dir is not None:
             return [stats_list]
         else:
-            return [state_dict, stats_list]
+            if self.rank == 0:
```

Comment:
I think we should also modify Line363 in pytorch_pyspark_estimator?
```
            self.state_dict = res[0]  # state dicts of all runners would be the same
            # Each runner would return a list of worker stats for different epochs
            worker_stats = [item for item in res if isinstance(item, list)]
```
rank0 worker's result doesn't necessarily comes first?